From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct inet6_dev
Patch-mainline: never, kabi
References: kabi

In 3.12.71, commit 6a4840575c9c7c43ee10ba105ccdbd77f9501a6e (ipv6:
simplify detection of first operational link-local address on
interface), upstream commit 11ffff752c6a5adc86f7dd397b2f75af8f917c51
removed valid_ll_addr_cnt from struct inet6_dev. It indeed changed the
layout of the structure.

So reintroduce the member as a void/unused int as it is used by the core
ipv6 networking code.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/if_inet6.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/net/if_inet6.h
+++ b/include/net/if_inet6.h
@@ -166,6 +166,7 @@ struct inet6_dev {
 	struct net_device	*dev;
 
 	struct list_head	addr_list;
+	int			valid_ll_addr_cnt; /* unused -- kABI */
 
 	struct ifmcaddr6	*mc_list;
 	struct ifmcaddr6	*mc_tomb;
